Apps/W1/Shopify/test/Catalogs/ShpfyCatalogPricesTest.Codeunit.al

```diff
@@ -72,4 +73,241 @@ codeunit 139646 "Shpfy Catalog Prices Test"
+        Item: Record Item;
+        CustomerDiscountGroup: Record "Customer Discount Group";
+        Customer: Record Customer;
+        InitializeTest: Codeunit "Shpfy Initialize Test";
+        ProductInitTest: Codeunit "Shpfy Product Init Test";
+        CatalogInitialize: Codeunit "Shpfy Catalog Initialize";
+        CompanyInitialize: Codeunit "Shpfy Company Initialize";
+        ProductPriceCalculation: Codeunit "Shpfy Product Price Calc.";
+        InitUnitCost: Decimal;
+        InitPrice: Decimal;
+        InitDiscountPerc: Decimal;
+        UnitCost: Decimal;
+        Price: Decimal;
+        ComparePrice: Decimal;
+    begin
+        // [GIVEN] Initializing test environment and creating necessary test records.
+        Shop := InitializeTest.CreateShop();
+        CompanyInitialize.CreateShopifyCompany(ShopifyCompany);
+        Catalog := CatalogInitialize.CreateCatalog(ShopifyCompany);
+        CatalogInitialize.CopyParametersFromShop(Catalog, Shop);
+        InitUnitCost := Any.DecimalInRange(10, 100, 1);
+        InitPrice := Any.DecimalInRange(2 * InitUnitCost, 4 * InitUnitCost, 1);
+        InitDiscountPerc := Any.DecimalInRange(5, 20, 1);
+        Item := ProductInitTest.CreateItem(Shop."Item Templ. Code", InitUnitCost, InitPrice);
+
+        // Creating a customer entry, though it is generic as discounts apply to all customers.
+        LibrarySales.CreateCustomer(Customer);
+
+        // [WHEN] Calculating initial prices without any discounts applied.
+        ProductPriceCalculation.SetShopAndCatalog(Shop, Catalog);
+        ProductPriceCalculation.CalcPrice(Item, '', '', UnitCost, Price, ComparePrice);
+
+        // [THEN] Confirm initial price calculations match expectations.
+        LibraryAssert.AreEqual(InitUnitCost, UnitCost, 'Initial unit cost should match expected.');
+        LibraryAssert.AreEqual(InitPrice, Price, 'Initial price should match expected before discount application.');
+
+        // [GIVEN] Updating the catalog to apply a universal discount to all customers.
+#if CLEAN23
+        ProductInitTest.CreateAllCustomerPriceList(Shop.Code, Item."No.", InitPrice, InitDiscountPerc);
+        Catalog."Customer No." := Customer."No.";
+        Catalog.Modify();
+
+        // [WHEN] Recalculating prices after applying the discount.
+        ProductPriceCalculation.SetShopAndCatalog(Shop, Catalog);
+        ProductPriceCalculation.CalcPrice(Item, '', '', UnitCost, Price, ComparePrice);
+
+        // [THEN] Validate the results reflect the universal discount.
+        LibraryAssert.AreEqual(InitUnitCost, UnitCost, 'Unit cost should remain consistent after discount application.');
+        LibraryAssert.AreEqual(InitPrice, ComparePrice, 'Compare price should reflect the original price prior to any discounts.');
+        LibraryAssert.AreNearlyEqual(InitPrice * (1 - InitDiscountPerc / 100), Price, 0.01, 'The final price should accurately reflect the applied discount for all customers.');
+#endif
```

Comment:
What is the reason behind this? Why do you have two different tests in one procedure?

Reply:
Also why CLEAN23?

Reply:
Hello @onbuyuka I followed the original test case and use the same process as Pricelists needs to be enabled for these tests.

This was the original test

#if not CLEAN23
        ProductInitTest.CreateSalesPrice(Shop.Code, Item."No.", InitPrice);
        CustomerDiscountGroup := ProductInitTest.CreateSalesLineDiscount(Shop.Code, Item."No.", InitDiscountPerc);
#else
        CustomerDiscountGroup := ProductInitTest.CreatePriceList(Shop.Code, Item."No.", InitPrice, InitDiscountPerc);
#endif

The original test case was run without applying any discounts. The next steps involve enabling the Pricelists and applying the relevant discounts as per the process.

Reply:
Okay I see, thank you for the explanation